src/wp-includes/class-wp-block-type.php

```diff
@@ -540,4 +563,30 @@ public function get_attributes() {
 			$this->attributes :
 			array();
 	}
+
+	/**
+	 * Get block variations.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array[]
+	 */
+	public function get_variations() {
+		if ( ! isset( $this->variations ) ) {
+			$this->variations = array();
+			if ( is_callable( $this->variation_callback ) ) {
+				$this->variations = call_user_func( $this->variation_callback );
+			}
+		}
```

Comment:
```suggestion
		if ( $this->variation_callback && is_callable( $this->variation_callback ) ) {
			$this->variations = call_user_func( $this->variation_callback );
			$this->variation_callback = null;
		}
```

Another way of doing this would like this.

Reply:
Looking at other feedback, but there maybe a benefit of existing solution, as it keeps `variation_callback`.

Reply:
@spacedmonkey I had given this a thought but then I was thinking what if the developer needs the original callback name in the filter provided below?

Reply:
Like in  https://github.com/WordPress/gutenberg/pull/54801, here the problem is that variations are built a bit early. In the same way, there may be a case where the developer wants to run the original callback but at a later stage.

Reply:
Fair point. Works for me.